### ISSUE TITLE
Nvim crashes when opening a python file in WSL

### ISSUE THREAD

**Issue by Muream** [neovim/neovim#20270]
### Neovim version (nvim -v)

0.7.2 and 0.8.0-dev

### Vim (not Nvim) behaves the same?

No, vim 9.0

### Operating system/version

Windows 11 (10.0.22621.521) with wsl 0.66.2.0

Here's the result of `wsl --version` on my end:
```
WSL version: 0.66.2.0
Kernel version: 5.15.57.1
WSLg version: 1.0.42
MSRDC version: 1.2.3401
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.521
```

### Terminal name/version

Windows Terminal Preview 1.16.2524.0

### $TERM environment variable

xterm-256color

### Installation

Package manager of each wsl distribution
For Ubuntu I tried with both the stable and unstable ppas

### How to reproduce the issue

Install and open a wsl session (I tried with Ubuntu, Debian and Arch).

Then:
```
touch asdf.py
nvim --clean
:e asdf.py
```

### Expected behavior

The file opens with no crash

### Actual behavior

Neovim crashes with this error
```
nvim: src/unix/core.c:271: uv__finish_close: Assertion `!(handle->flags & UV_HANDLE_CLOSED)' failed.
[1]    3370 IOT instruction  nvim --clean asdf.py
```

**Comment by clason** [comment]
This is a crash in LibUV; maybe check their issue tracker for similar reports involving closing handles?

**Comment by Muream** [comment]
The only issue that comes up with the `wsl` and `handle` keywords is this one, not sure if it's really relevant
https://github.com/libuv/libuv/issues/2847

**Comment by clason** [comment]
Hmm. Then we need a [stacktrace](https://github.com/neovim/neovim/wiki/FAQ#debug) to see what is using this function (potentially incorrectly).

**Comment by Muream** [comment]
So, I couldn't follow the exact steps you linked to because apparently wsl doesn't start systemd
I followed a guide to enable it and neovim started working

I don't have a stack trace to provide as `coredumpctl -1 gdb` does result in `No match found` now.

I'm happy to close this unless you want to dig more

**Comment by clason** [comment]
Sorry, I can't do anything since it only crashes for you -- or maybe others, who may come across this and help out with a backtrace.

**Comment by Muream** [comment]
No problem at all, I've sort of fixed the issue anyway and sounds like it's more a wsl thing than a neovim issue anyway, I'll close the issue, hopefully it can be useful to others who run into the same problem

Thanks for your help!

**Comment by Muream** [comment]
For anyone reading this, Microsoft added support for systemd in WSL a couple hours after I closed this
You simply need to add these lines to `/etc/wsl.conf` in each wsl distro
```
[boot]
systemd=true
```
And then run `wsl.exe --shutdown` in powershell to restart the WSL instances

Here's more info on that: https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/

**Comment by detteiu8383** [comment]
I had the same problem in my environment (only some versions are different from @Muream's), but I was able to solve it by preventing the Windows environment variables from being added to WSL.

In my case, the [pyenv](https://github.com/pyenv/pyenv) environment variables set on the Windows side were also being referenced on the WSL side, causing an error when trying to open a file that did not exist. To fix this, I added the following lines to `/etc/wsl.conf`:
```conf
[interop]
appendWindowsPath = false
```
This prevents Windows environment variables from being added to the WSL side, eliminating the potential for unexpected errors. (See <https://learn.microsoft.com/en-us/windows/wsl/wsl-config#interop-settings> for more information.)
